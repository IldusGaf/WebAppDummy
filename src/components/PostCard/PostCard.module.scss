@import '../styleConstants.scss';
.postCard {
    display: flex;
    background-color: $elementColor;
    border: 0.1rem solid $elementBorderColor;
    flex-direction: column;
    justify-content: center;
    border-radius: $contentBorderRadius;
    margin: 0.5rem;
    padding: 1rem;
    box-sizing: content-box;
    &__user {
        display: flex;
        align-items: center;
        margin-bottom: 0.4rem;
    }
    &__userImage {
        margin: 0;
    img {
        border-radius: 50%;
        height: 40px;
        width: 40px;
    }
}
    &__userDesc {
        display: flex;
        flex-direction: column;
        margin-left: 0.5rem;
    }
    &__userName {
        display: flex;
    }
    &__userDate {
        color: gray;
    }
    &__image {
        object-fit: cover;
        background-color: $backColor;
        border-radius: $contentBorderRadius;
        margin-bottom: 0.3rem;
        @include mediaSizeImg;
    }
    &__desc {
        color: inherit;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }
    &Dark{
        @include darkContent;
    }
    @include mediaSizeCards;
}