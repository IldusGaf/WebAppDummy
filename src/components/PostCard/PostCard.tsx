// import { UserOutlined } from '@ant-design/icons';
// import { Avatar } from 'antd';
import React, { useContext } from 'react';
import { ThemeContext } from '../../contexts/ThemeContext';
import { PostType } from '../../types/dummyAPIResponses';
import classes from './PostCard.module.scss';

interface Props extends PostType {
  visibleOwner?: boolean
}

export const PostCard = ({
  text,
  image,
  owner,
  publishDate,
  visibleOwner,
}:Props) => {
  const themeContext = useContext(ThemeContext);
  return (
    <div className={`${classes.postCard} ${themeContext.darkTheme ? classes.postCardDark : ''}`}>
      {!visibleOwner && (
      <div className={classes.postCard__user}>
        {/* <Avatar icon={<UserOutlined />} size="large" /> */}
        <figure className={classes.postCard__userImage}><img alt={owner.firstName} src={owner.picture} /></figure>
        <div className={classes.postCard__userDesc}>
          <span className={classes.postCard__userName}>
            {owner.firstName}
            {' '}
            {owner.lastName}
          </span>
          <span className={classes.postCard__userDate}>{publishDate}</span>
        </div>
      </div>
      )}
      <img className={classes.postCard__image} alt="" src={image} />
      <span className={classes.postCard__desc}>{text}</span>
    </div>
  );
};
