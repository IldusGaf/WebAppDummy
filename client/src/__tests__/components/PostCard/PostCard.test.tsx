import React from 'react';
import { shallow } from 'enzyme';
import { PostCard } from '../../../components/PostCard/PostCard';

describe('PostCard component test', () => {
  test('should render PostCard', () => {
    const wrapper = shallow(<PostCard image="" text="" owner={{ firstName: '', lastName: '' }} publishDate="" />);
    expect(wrapper.find('div.postCard')).toHaveLength(1);
  });

  test('should render data', () => {
    const title = 'mr';
    const firstName = 'James';
    const lastName = 'Bond';
    const publishDate = '12.06.2020';
    const text = 'Blah-blah-blah';
    const picture = 'pictureUrl';
    const wrapper = shallow(<PostCard
      owner={{ firstName, lastName, picture }}
      publishDate={publishDate}
      text={text}
      image={picture}
    />);
    expect(wrapper.find('span.postCard__userName').text()).toBe(`${firstName} ${lastName}`);
    expect(wrapper.find('span.postCard__userDate').text()).toBe('06 December 2020 00:00');
    expect(wrapper.find('span.postCard__desc').text()).toBe(text);
  });

  test('snapshot test', () => {
    const wrapper = shallow(<PostCard image="" text="" owner={{ firstName: '', lastName: '', picture: '' }} publishDate="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
