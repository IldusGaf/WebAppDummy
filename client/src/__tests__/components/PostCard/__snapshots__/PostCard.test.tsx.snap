// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard component test snapshot test 1`] = `
<div
  className="postCard "
>
  <div
    className="postCard__user"
  >
    <figure
      className="postCard__userImage"
    >
      <img
        alt=""
        src=""
      />
    </figure>
    <div
      className="postCard__userDesc"
    >
      <span
        className="postCard__userName"
      >
         
      </span>
      <span
        className="postCard__userDate"
      />
    </div>
  </div>
  <img
    alt=""
    className="postCard__image"
    src=""
  />
  <span
    className="postCard__desc"
  />
</div>
`;
